src/test/java/teammates/ui/webapi/GetNotificationActionTest.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    @Override
+    String getRequestMethod() {
+        return GET;
+    }
+
+    @Test
+    @Override
+    protected void testExecute() {
+        // See independent test cases
+    }
+
+    @Test
+    @Override
+    protected void testAccessControl() {
+        InstructorAttributes instructor = typicalBundle.instructors.get("instructor1OfCourse1");
+        loginAsInstructor(instructor.getGoogleId());
+        ______TS("student notification not accessible to instructor");
+        String[] requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, NotificationTargetUser.STUDENT.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
+        };
+        verifyCannotAccess(requestParams);
+
+        ______TS("accessible to instructor");
+        requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, NotificationTargetUser.INSTRUCTOR.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
+        };
+        verifyCanAccess(requestParams);
+
+        ______TS("instructor notification not accessible to student");
+        StudentAttributes studentAttributes = typicalBundle.students.get("student1InCourse1");
+        loginAsStudent(studentAttributes.getGoogleId());
+        requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, NotificationTargetUser.INSTRUCTOR.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
+        };
+        verifyCannotAccess(requestParams);
+
+        ______TS("accessible to student");
+        requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, NotificationTargetUser.STUDENT.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
+        };
+        verifyCanAccess(requestParams);
+
+        ______TS("unknown target user");
+        loginAsInstructor(instructor.getGoogleId());
+        requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, "unknown",
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
+        };
+        verifyHttpParameterFailureAcl(requestParams);
+
+        ______TS("accessible to admin");
+        loginAsAdmin();
+        requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_TARGET_USER, NotificationTargetUser.STUDENT.toString(),
+                Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL, String.valueOf(true),
+        };
+        verifyCanAccess(requestParams);
+    }
+
+    @Test
+    public void testExecute_withValidUserTypeForNonAdmin_shouldReturnData() {
+        ______TS("Request to fetch notification");
+        int expectedNumberOfNotifications = 4;
+        InstructorAttributes instructor = typicalBundle.instructors.get("instructor1OfCourse1");
```

Comment:
The expected number of active notifications here is dependent on the current time, we are in 2022 but two years later the end time for some of these notifications e.g. `notification-4` would have passed, and this test will fail 😄 . 

```
    "notification-4": {
      "notificationId": "notification-4",
      "startTime": "2011-04-04T00:00:00Z",
      "endTime": "2024-04-04T00:00:00Z",
      "createdAt": "2011-01-01T00:00:00Z",
      "updatedAt": "2011-01-01T00:00:00Z",
      "type": "MAINTENANCE",
      "targetUser": "GENERAL",
      "title": "The note of maintenance",
      "message": "The content of maintenance",
      "shown": false
    },
```

Reply:
Would it be okay if I change the endtime to for example 2099?

Reply:
> Would it be okay if I change the endtime to for example 2099?

This has been updated in the newly pushed code. Thank you!

Reply:
We will leave it like this for now, although I think there's a better way to do it which involve changing the code in `NotificationsDb`.